### DISCUSSION TITLE
CPU Cycle of Renovate

### DISCUSSION THREAD

**Discussion by bachtran69** [renovatebot/renovate#16512] [Request Help]
My organization are considering using GitHub App Renovate for our project. We would like to know more about the CPU Cycle of Renovate, especially how our codebase are read and how it can be secured. 

Could anyone assist me with this question? Thank you in advance!

**Comment by HonkingGoose** [comment]
> We would like to know more about the CPU Cycle of Renovate, especially how our codebase are read and how it can be secured.

First, read the [Renovate docs, Security and Permissions](https://docs.renovatebot.com/security-and-permissions/). It explains _what_ permissions Renovate needs, and _why_.

I don't know anything about the "CPU Cycle of Renovate", I'll let the maintainers answer this.

**Comment by rarkins** [comment]
As a way to optimize how much we can invest in this open source, we don't spend much time/money answering security questions about a free product or service. We understand that some companies may choose not to use the app unless they can cross-examine a vendor, so in such cases instead they can self-host and inspect the source code.
